frontend/src/app/Catalog/CatalogView/CatalogTabs.tsx

```diff
@@ -0,0 +1,134 @@
+	loading: boolean;
+	abbreviation: string;
+	courseNumber: string;
+}
+
+const CatalogTabs = (props: CatalogTabsProps) => {
+	const { semester, course, sections, loading, abbreviation, courseNumber } = props;
+
+	const [hoveredClass, setHoveredClass] = useState<boolean>(false);
+	const [updateMobileHover, setUpdateMobileHover] = useState<boolean>(true);
+	const [tab, setTab] = useState<number>(0);
+
+	const gradesGraphData = useSelector((state: State) => state.grade.graphData ?? null);
+	const gradesData = useSelector((state: State) => state.grade?.gradesData ?? null);
+	const selectedCourses = useSelector((state: State) => state.grade.selectedCourses ?? null);
+
+	const links: string[] = catalogService.getLinks(sections, semester, abbreviation, courseNumber);
+
+	function update(course: CourseFragment, grade: GradeType) {
+		if (course && gradesData && gradesData.length > 0) {
+			const selectedGrades = gradesData.filter((c: CourseFragment) => course.id === c.id)[0];
+			const hoverTotal = {
+				...course,
+				...selectedGrades,
+				hoverGrade: grade
+			};
+
+			setHoveredClass(hoverTotal);
+		}
+	}
+
+	// Handler function for updating GradesInfoCard on hover with single course
+	function updateGraphHover(data: any) {
+		const { isTooltipActive, activeLabel } = data;
+		const noBarMobile = updateMobileHover && window.innerWidth < 768;
+
+		// Update the selected course if no bar is clicked if in mobile
+		if (isTooltipActive && (selectedCourses.length === 1 || noBarMobile)) {
+			const selectedCourse = selectedCourses[0];
+			const grade = activeLabel;
+			update(selectedCourse, grade);
+		}
+
+		// Update mobile hover records if there actually is a bar (then we only want updateBarHover to run)
+		setUpdateMobileHover({ updateMobileHover: true });
+	}
+
+	const renderTabs = (currentTab: number) => {
+		if (loading) {
+			return <BTLoader />;
+		}
+
+		switch (currentTab) {
+			case 0:
+				return (
+					<div className={styles.gradesBox}>
+						{sections && sections.length > 0 ? (
+							<SectionTable links={links} sections={sections} />
+						) : (
+							'There are no class times for the selected course.'
+						)}
+					</div>
+				);
+
+			case 1:
+				return gradesData?.length > 0 ? (
+					<div className={styles.gradesBox}>
+						<GradesGraph
+							graphData={gradesGraphData}
+							gradesData={gradesData}
+							updateBarHover={null}
+							updateGraphHover={updateGraphHover}
+							course={course?.abbreviation + ' ' + course?.courseNumber}
+							semester={'All Semesters'}
+							instructor={'All Instructors'}
+							selectedPercentiles={hoveredClass[hoveredClass.hoverGrade]}
+							denominator={hoveredClass.denominator}
+							color={0}
+							isMobile={window.innerWidth < 768 ? true : false}
+							graphEmpty={false}
+						/>
+					</div>
+				) : (
+					<div>no grade data found</div>
+				);
+			case 2:
+				return (
+					<div className={styles.gradesBox}>
+						<EnrollmentGraphCard
```

Comment:
Shows "You have not added any classes yet." when no data is available. Remove graph and show user-friendly message when enrollment data isn't available like for grades data

Reply:
Maybe (?), noticed grades data tooltip wasn't updating when hovering for some classes